src/parser/warlock/demonology/modules/pets/PetTimelineTab/TabComponent/index.js

```diff
@@ -1,34 +1,126 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 
-import Panel from 'interface/others/Panel';
-
+import CombatLogParser from 'parser/core/CombatLogParser';
+import DragScroll from 'interface/common/DragScroll';
+import { formatDuration } from 'common/format';
 import SpellLink from 'common/SpellLink';
-import SPELLS from 'common/SPELLS';
+import Icon from 'common/Icon';
+
+import PetTimelineProcesser from './PetTimelineProcesser';
+import KeyCastsRow from './KeyCastsRow';
+import PetRow from './PetRow';
+import './Timeline.scss';
 
-import PetTimeline from './PetTimeline';
+const ZOOM = 2;
 
-class TimelineTab extends React.PureComponent {
+class TimelineTab extends React.Component {
   static propTypes = {
-    selectedCombatant: PropTypes.object,
+    parser: PropTypes.instanceOf(CombatLogParser).isRequired,
   };
 
+  constructor(props) {
+    super(props);
+    this.state = {
+      timelineHeight: 50,
+    };
+    this.setHeight = this.setHeight.bind(this);
+  }
+
+  get fight() {
+    return this.props.parser.fight;
+  }
+
+  get start() {
+    return this.fight.start_time;
+  }
+
+  get end() {
+    return this.fight.end_time;
+  }
+
+  get duration() {
+    return this.end - this.start;
+  }
+
+  get seconds() {
+    return this.duration / 1000;
+  }
+
+  get secondWidth() {
+    return 120 / ZOOM;
+  }
+
+  get totalWidth() {
+    return this.seconds * this.secondWidth;
+  }
+
+  get skipInterval() {
+    return Math.ceil(40 / this.secondWidth);
+  }
+
+  setHeight(element) {
+    if (!element || !element.clientHeight) {
+      return;
+    }
+    this.setState({
+      timelineHeight: element.clientHeight,
+    });
+  }
+  // shamelessly copied from new Timeline
```

Comment:
I am not really experienced in javascript, would it be possible to extend Timeline? so you don't have to copy these functions?

Reply:
Extending React components *is* a possibility. Though here, the only thing that's identical is the style. The data the timeline is based on is completely different from the spell timeline so I don't think I can extend it.